lib/crypt/ogs-kdf.c

```diff
@@ -181,33 +181,35 @@ void ogs_kdf_kseaf(char *serving_network_name, uint8_t *kausf, uint8_t *kseaf)
 }
 
 /* TS33.501 Annex A.7 : Kamf derivation function */
-void ogs_kdf_kamf(char *supi, uint8_t *abba, uint8_t abba_len,
-        uint8_t *kseaf, uint8_t *kamf)
+void ogs_kdf_kamf(const char *supi, const uint8_t *abba, uint8_t abba_len,
+        const uint8_t *kseaf, uint8_t *kamf)
 {
     kdf_param_t param;
+    char *val;
 
     ogs_assert(supi);
     ogs_assert(abba);
     ogs_assert(abba_len);
     ogs_assert(kseaf);
     ogs_assert(kamf);
 
+    val = ogs_id_get_value(supi);
```

Comment:
FYI, only change other than adding "const" keywords is this one, in case you want to review.

Reply:
I changed it because otherwise "ogs_free(param[0].buf);" triggers a compiler warning about freeing a const pointer.